addons/l10n_se/models/res_company.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+import re
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    org_number = fields.Char(compute='_compute_org_number')
+
+    @api.depends('vat')
+    def _compute_org_number(self):
+        for company in self:
+            if company.country_id == self.env.ref('base.se'):
+                org_number = re.sub(r'\D', '', company.vat)[:-2]
+                org_number = org_number[:6] + '-' + org_number[6:]
+
+                company.org_number = org_number
```

Comment:
What happens when we reset the VAT to an empty field? What should org_number contain? Shouldn't it be empty?

Reply:
You can't report vat if you don't have a vat number, so is it a problem ? Am I supposed to play defensive there and set an empty orgnumber if no vat is provided ?

Reply:
I checked the condition and put an empty orgnumber if there is no vat

Reply:
> I checked the condition and put an empty orgnumber if there is no vat

This should be the correct way. 

Some quick information about organistional numbers that might be good to know. The first digit in the organisational number tells you what type of company it is like this:

2: stat, kommun eller landsting
5: aktiebolag
7: ekonomisk förening
8: stiftelse eller ideell förening
9: handels- eller kommanditbolag

This is the case in all except enskild firma, that company type uses the personal number as the organisational number.